frontend/src/components/cfdtables/CfdTable.tsx

```diff
@@ -0,0 +1,271 @@
+import { ChevronRightIcon, ChevronUpIcon, TriangleDownIcon, TriangleUpIcon } from "@chakra-ui/icons";
+import { Badge, Box, chakra, HStack, IconButton, Table as CUITable, Tbody, Td, Th, Thead, Tr } from "@chakra-ui/react";
+import React from "react";
+import { Column, Row, useExpanded, useSortBy, useTable } from "react-table";
+import { Cfd } from "../Types";
+
+interface CfdTableProps {
+    data: Cfd[];
+}
+
+export function CfdTable(
+    { data }: CfdTableProps,
+) {
+    const tableData = React.useMemo(
+        () => data,
+        [data],
+    );
+
+    const columns: Array<Column<Cfd>> = React.useMemo(
+        () => [
+            {
+                id: "expander",
+                Header: () => null,
+                Cell: ({ row }: any) => (
+                    <span {...row.getToggleRowExpandedProps()}>
+                        {row.isExpanded
+                            ? <IconButton
+                                aria-label="Reduce"
+                                icon={<ChevronUpIcon />}
+                                onClick={() => {
+                                    row.toggleRowExpanded();
+                                }}
+                            />
+                            : <IconButton
+                                aria-label="Expand"
+                                icon={<ChevronRightIcon />}
```

Comment:
nit: I think this should be a `ChevronDownIcon` to indicate that this can be "unfolded downwards".

Reply:
I wasn't sure, it looked weird when I looked at it :D

Reply:
For me it feels weirder at the moment, because for me the right chevron indicates a link that takes me somewhere else. 
I think down is correct, but we can tackle that later :)

Reply:
I'll leave that up to you ok? No strong feeling on this one :)